embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpansService.kt

```diff
@@ -25,15 +25,16 @@ internal class EmbraceSpansService(
         if (!initialized()) {
             synchronized(currentDelegate) {
                 if (!initialized()) {
-                    currentDelegate = SpansServiceImpl(
+                    val realSpansService = SpansServiceImpl(
                         spansRepository = spansRepository,
                         currentSessionSpan = currentSessionSpan,
                         tracer = tracerSupplier(),
                     )
-                    currentDelegate.initializeService(sdkInitStartTimeNanos)
-                    if (currentDelegate.initialized()) {
-                        uninitializedSdkSpansService.recordBufferedCalls(this)
+                    realSpansService.initializeService(sdkInitStartTimeNanos)
+                    if (realSpansService.initialized()) {
+                        uninitializedSdkSpansService.triggerBufferedSpanRecording(realSpansService)
```

Comment:
Init fix when the wrapper was being used rather than the real working SpansService

Reply:
Nice catch - that's a tricky state-related bug to spot

Reply:
The buffering fix might actually fix this too, but I did this earlier and it seems to be more correct anyway. Why use the wrapper when you know exactly which instance you should use.